provider/datagen/src/transform/cldr/currency/mod.rs

```diff
@@ -0,0 +1,346 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_locid::LanguageIdentifier;
+use icu_properties::sets::load_for_general_category_group;
+use icu_properties::GeneralCategoryGroup;
+use icu_provider::DataProvider;
+use tinystr::UnvalidatedTinyAsciiStr;
+use zerovec::VarZeroVec;
+use zerovec::ZeroMap;
+
+use crate::transform::cldr::cldr_serde;
+use crate::DatagenProvider;
+use icu_provider::datagen::IterableDataProvider;
+use icu_provider::prelude::*;
+use icu_singlenumberformatter::provider::*;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use tinystr::tinystr;
+
+/// Returns the pattern selection for a currency.
+/// For example:
+///    if the pattern is ¤#,##0.00 and the symbol is EGP,
+///    this means the return value will be PatternSelection::StandardAlphaNextToNumber
+///    because the character closes to the number is a letter.
+fn currency_pattern_selection(
+    provider: &DatagenProvider,
+    pattern: String,
+    place_holder: String,
+) -> Result<PatternSelection, DataError> {
+    // TODO(younies): what should we do when the place_holder is empty?
+    if place_holder.is_empty() {
+        todo!("Handle empty place holders")
+    }
+
+    let currency_sign = '¤';
+    let currency_sign_index = pattern.find(currency_sign).unwrap();
+    let first_num_index = pattern.find(|c| c == '0' || c == '#').unwrap();
+    let last_num_index = pattern.rfind(|c: char| c == '0' || c == '#').unwrap();
+
+    let letters_set = match load_for_general_category_group(provider, GeneralCategoryGroup::Letter)
+    {
+        Ok(letters_set) => letters_set,
+        Err(icu_properties::PropertiesError::PropDataLoad(e)) => {
+            return Err(e);
+        }
+        Err(_) => unreachable!("load_for_general_category_group should only return PropDataLoad"),
+    };
+
+    let char_closer_to_number = {
+        if currency_sign_index < first_num_index {
+            place_holder.chars().last().unwrap()
```

Comment:
nit: use `.next_back()` or `.rev().next()`. `.last()` iterates all the way from the front.